### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by d5c5ceb0**
## Summary by CodeRabbit

- **New Features**
	- Introduced a command-line interface (CLI) for managing server operations.
	- Added support for zero-knowledge proof generation and verification via new API endpoints.
	- Implemented a logging system for tracking application activities.
	- New neural network models and associated configuration files for enhanced functionality.
	- Added configuration file `config.yaml` for server, database, and public resource settings.

- **Documentation**
	- Updated README.md with project overview, features, installation instructions, and usage guidelines.

- **Bug Fixes**
	- Improved error handling for server requests and configuration loading.

- **Chores**
	- Added configuration files for server and database settings.
	- Introduced new JSON configuration files for model execution parameters.

### CODE REVIEW

File: models/1l_average/gen.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+from ezkl import export
+
+
+class Model(nn.Module):
+    def __init__(self):
+        super(Model, self).__init__()
+        self.layer = nn.AvgPool2d(2, 1, (1, 1))
+
+    def forward(self, x):
+        return self.layer(x)[0]
```
[the remark this answers is not in the document]
Comment (d5c5ceb0) [reply models/1l_average/gen.py:11]: model file, not code

File: models/1l_average/gen.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+
+circuit = Model()
+export(circuit, input_shape=[3, 2, 2])
```
[the remark this answers is not in the document]
Comment (d5c5ceb0) [reply models/1l_average/gen.py:15]: model file, not code

File: models/2l_relu_fc/input.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+            0.07870037853717804,
+            0.08462252467870712
+        ]
+    ],
+    "output_data": [
+        [
+            0.0
+        ]
+    ]
+}
```
[the remark this answers is not in the document]
Comment (d5c5ceb0) [reply models/2l_relu_fc/input.json:20]: model file, not code

File: models/2l_relu_fc/gen.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        self._initialize_weights()
+
+    def forward(self, x):
+        x =  self.aff1(x)
+        x =  self.relu(x)
+        return (x)
+
+
+    def _initialize_weights(self):
+        init.orthogonal_(self.aff1.weight)
```
[the remark this answers is not in the document]
Comment (d5c5ceb0) [reply models/2l_relu_fc/gen.py:20]: model file, not code

File: models/2l_relu_fc/gen.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import torch.nn.init as init
+from ezkl import export
+
+class Model(nn.Module):
+    def __init__(self, inplace=False):
+        super(Model, self).__init__()
+
+        self.aff1 = nn.Linear(3,1)
+        self.relu = nn.ReLU()
+        self._initialize_weights()
```
[the remark this answers is not in the document]
Comment (d5c5ceb0) [reply models/2l_relu_fc/gen.py:11]: model file, not code

**Review by bufrr — APPROVED** [review]

File: src/handler/handler.rs [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+        req_id: req.req_id,
+        code: 200,
+        result: res.to_string(),
+    })
+}
+
+async fn fetch_to_temp_file(url: &str) -> ZKMLResult<NamedTempFile> {
+    let response = reqwest::get(url)
+        .await
+        .map_err(|e| ZKMLError::OtherError(e.to_string()))?;
```
[the remark this answers is not in the document]
Comment (d5c5ceb0) [reply src/handler/handler.rs:161]: deal with request timeout in run_cli().

File: src/cli/mod.rs [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    let mut help_info = true;
+    let args = OperatorCli::from_args();
+
+    if let Some(config_path) = args.config_path {
+        help_info = false;
+
+        let cors = CorsLayer::new()
+            .allow_origin(Any)
+            .allow_methods([Method::GET, Method::POST])
+            .allow_headers(Any);
```
[the remark this answers is not in the document]
Comment (d5c5ceb0) [reply src/cli/mod.rs:44]: allow_origin is not determined.

File: src/handler/.handler.rs.swp
Comment (bufrr) [inline src/handler/.handler.rs.swp:1]: vim cache?

File: src/handler/handler.rs [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+        result: "healthy".to_string(),
+    })
+}
+
+#[debug_handler]
+pub async fn prove(
+    State(state): State<SharedState>,
+    Json(req): Json<ProveRequest>,
+) -> Json<Response<String>> {
+    let _lock = state.0.write().await;
```
Comment (bufrr) [inline src/handler/handler.rs:44]: You don't need to hold the lock, which will cause a concurrent issue.

File: src/handler/handler.rs [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+}
+
+#[debug_handler]
+pub async fn verify(
+    State(state): State<SharedState>,
+    Json(req): Json<VerifyRequest>,
+) -> Json<Response<String>> {
+    tracing::info!("verify request: {:?}", &req);
+
+    let _lock = state.0.write().await;
```
Comment (bufrr) [inline src/handler/handler.rs:86]: Copy the value then drop?

**Pull request merged**